pkg/volume/image/image.go

```diff
@@ -44,9 +45,16 @@ func ProbeVolumePlugins() []volume.VolumePlugin {
 	return []volume.VolumePlugin{p}
 }
 
-func (o *imagePlugin) Init(volume.VolumeHost) error                    { return nil }
-func (o *imagePlugin) GetPluginName() string                           { return pluginName }
-func (o *imagePlugin) GetVolumeName(spec *volume.Spec) (string, error) { return o.spec.Name(), nil }
+func (o *imagePlugin) Init(volume.VolumeHost) error { return nil }
+func (o *imagePlugin) GetPluginName() string        { return pluginName }
+func (o *imagePlugin) GetVolumeName(spec *volume.Spec) (string, error) {
+	volumeSource := getVolumeSource(spec)
+	if volumeSource == nil {
+		return "", fmt.Errorf("the volumeSpec does not reference an Image volume type")
+	}
+
+	return volumeSource.Reference, nil
+}
```

Comment:
The spec field is not initialized in the code. If the function is called, the kubelet will panic. /cc @saschagrunert 

DSW: 
- For attachable/device-mountable volumes, use the unique volume name as reported by the plugin.
- For non-attachable and non-device-mountable volumes, generate a unique name based on the pod namespace and name and the name of the volume within the pod.

The image plugin almost acts as a noop plugin except for the `CanSupport` func, so the `GetVolumeName` funcution is not called. I don't know whether the function will be never called in the feature.

There are two different ways to generate volume name:

1. configmap plugin: combine the configmap name and outer volume name.
2. secret plugin: use the secret name only.

I am confused how to generate the volume name for this plugin.  /cc @msau42 @jsafrane

Reply:
/cc @gnufied